Assets/_BForBoss/_Core/Scripts/Labor/LaborSystem.cs

```diff
@@ -0,0 +1,53 @@
+using System;
+using System.Collections.Generic;
+
+namespace BForBoss.Labor
+{
+    public class LaborSystem
+    {
+        public Action onLaborCompleted;
+        public Action onQueueCompleted;
+        
+        private Queue<ILabor> _laborsToComplete;
+        public ILabor CurrentLabor { get; private set; }
+
+        public void Initialize(IEnumerable<ILabor> labors)
```

Comment:
If we're not using Mono, this could be an actual constructor

Reply:
Yeah I wasn't sure if we wanted to couple the labor system to a specific set of labors. I thought this seemed more reusable, ie one labor system that gets initialized multiple times with different sets of labors. I might be overcomplicating it though

Reply:
Yeah I wasn't sure if we wanted to couple the labor system to a specific set of labors. I thought this seemed more reusable, ie one labor system that gets initialized multiple times with different sets of labors. I might be overcomplicating it though